adding debug info to Dev guide (#1881)
temporalio/documentation: docs-src/app-dev-context/ide-debug.md

> ---

Reviewer: The document should mention this applies to workers _and_ replayers (even though it's kind of implied to us). Also, you might want to mention that debugging combines very well with "replaying" (and combine with that doc if/when ready) to step through past executions.

id: ide-debug
title: Debug with IDE
description: How to debug your code using your favorite IDE
sidebar_label: Debug with IDE
tags:
  - guide-context
---

Many IDEs support a debugging feature that enables you to step through your code.

> You can use these tools with your Workflow code, however make sure to set the `TEMPORAL_DEBUG` environment variable to `true` first.

Reviewer 2: ```suggestion
You can use these tools with your Workflow code; however, make sure to set the `TEMPORAL_DEBUG` environment variable to `true` first.
```

> You can use these tools with your Workflow code, however make sure to set the `TEMPORAL_DEBUG` environment variable to `true` first.

Reviewer: Python also allows you to set `debug_mode=True` and TypeScript also allows `debugMode: true` as alternatives to environment variables.

Reviewer: In either the replayer or worker on each. So for TypeScript it's [debugInfo](https://typescript.temporal.io/api/interfaces/worker.WorkerOptions#debugmode) on `Worker.create` (worker and replayer combined in that language). For Python, it's `debug_mode` on [Worker.__init__](https://python.temporal.io/temporalio.worker.Worker.html#__init__) or [Replayer.__init__](https://python.temporal.io/temporalio.worker.Replayer.html#__init__).

If the env var is set, it will be the same as setting those to true in Python and [soon in TypeScript](https://github.com/temporalio/sdk-typescript/issues/1020).

Reviewer 3: Slight correction, it's `debugMode` for TS.

> The SDKs include deadlock detection which fails a Workflow Task whenever the code blocks for over a second without relinquishing execution control.

Reviewer 2: ```suggestion
The SDKs include deadlock detection, which fails a Workflow Task whenever the code blocks for more than a second without relinquishing execution control.
```

> If the environment variable is not set, you can often encounter a "potential deadlock detected" error while stepping through Workflow Definitions during debugging.

Reviewer: ```suggestion
If the environment variable is not set, you can often encounter a deadlock detected error while stepping through Workflow Definitions during debugging.
```

Should not assume what the error messages are in all languages.

Reviewer 2: Good catch. In this usage, I recommend hyphenating "deadlock-detected".
